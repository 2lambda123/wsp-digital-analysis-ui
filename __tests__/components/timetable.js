/* global describe, it, expect, jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Timetable from '../../lib/components/timetable'

describe('Timetable', () => {
  it('renders correctly', () => {
    const timetable = {
      name: 'Test timetable',
      speed: 40,
      dwellTime: 10,
      monday: true,
      tuesday: true,
      wednesday: true,
      thursday: true,
      friday: true,
      saturday: false,
      sunday: false,
      headwaySecs: 900, // 15 minutes
      startTime: 28800, // 8am
      endTime: 57600 // 4pm
    }
    const removeTimetableFn = jest.fn()
    const replaceTimetableFn = jest.fn()
    const tree = renderer.create(
      <Timetable
        timetable={timetable}
        removeTimetable={removeTimetableFn}
        replaceTimetable={replaceTimetableFn}
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    expect(removeTimetableFn).not.toBeCalled
    expect(replaceTimetableFn).not.toBeCalled
  })
})
