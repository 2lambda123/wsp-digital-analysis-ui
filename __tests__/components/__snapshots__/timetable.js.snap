exports[`Component > Timetable renders correctly 1`] = `
<Timetable
  removeTimetable={[Function]}
  replaceTimetable={[Function]}
  timetable={
    Object {
      "dwellTime": 10,
      "endTime": 57600,
      "friday": true,
      "headwaySecs": 900,
      "monday": true,
      "name": "Test timetable",
      "patternTrips": Array [
        "abcd",
      ],
      "saturday": false,
      "sourceTrip": "abcd",
      "speed": 40,
      "startTime": 28800,
      "sunday": false,
      "thursday": true,
      "tuesday": true,
      "wednesday": true,
    }
  }>
  <section
    className="panel panel-default inner-panel">
    <div
      className="panel-heading clearfix">
      <strong>
        Test timetable
      </strong>
      <pure(Component)
        className="pull-right"
        onClick={[Function]}
        size="sm"
        style="danger"
        title="Delete timetable">
        <a
          className="btn btn-danger btn-sm pull-right"
          href="#"
          onClick={[Function]}
          target={undefined}
          title="Delete timetable">
          <pure(Icon)
            type="close">
            <i
              className="fa fa-close fa-fw " />
          </pure(Icon)>
        </a>
      </pure(Component)>
    </div>
    <div
      className="panel-body">
      <pure(Component)
        name="Name"
        onChange={[Function]}
        value="Test timetable">
        <Group
          name="Name"
          onChange={[Function]}
          value="Test timetable">
          <div
            className="form-group">
            <pure(Component)
              name="Name"
              onChange={[Function]}
              type="text"
              value="Test timetable">
              <input
                className="form-control"
                defaultValue="Test timetable"
                name="Name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
                value="Test timetable" />
            </pure(Component)>
          </div>
        </Group>
      </pure(Component)>
      <TimetableEntry
        replaceTimetable={[Function]}
        timetable={
          Object {
            "dwellTime": 10,
            "endTime": 57600,
            "friday": true,
            "headwaySecs": 900,
            "monday": true,
            "name": "Test timetable",
            "patternTrips": Array [
              "abcd",
            ],
            "saturday": false,
            "sourceTrip": "abcd",
            "speed": 40,
            "startTime": 28800,
            "sunday": false,
            "thursday": true,
            "tuesday": true,
            "wednesday": true,
          }
        }>
        <div>
          <Days
            setDay={[Function]}
            timetable={
              Object {
                "dwellTime": 10,
                "endTime": 57600,
                "friday": true,
                "headwaySecs": 900,
                "monday": true,
                "name": "Test timetable",
                "patternTrips": Array [
                  "abcd",
                ],
                "saturday": false,
                "sourceTrip": "abcd",
                "speed": 40,
                "startTime": 28800,
                "sunday": false,
                "thursday": true,
                "tuesday": true,
                "wednesday": true,
              }
            }>
            <div
              className="form-inline">
              <pure(Component)
                checked={true}
                label="Mon"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={true}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={true}
                label="Tue"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={true}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={true}
                label="Wed"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={true}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={true}
                label="Thu"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={true}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={true}
                label="Fri"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={true}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={false}
                label="Sat"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
              <pure(Component)
                checked={false}
                label="Sun"
                onChange={[Function]}>
                <div
                  className="checkbox">
                  <label>
                    <input
                      checked={false}
                      onChange={[Function]}
                      type="checkbox" />
                  </label>
                </div>
              </pure(Component)>
            </div>
          </Days>
          <pure(Component)
            label="Frequency (minutes)"
            onChange={[Function]}
            value={15}>
            <Group
              label="Frequency (minutes)"
              onChange={[Function]}
              value={15}>
              <div
                className="form-group">
                <label>
                  Frequency (minutes)
                </label>
                <pure(Component)
                  label="Frequency (minutes)"
                  onChange={[Function]}
                  type="number"
                  value={15}>
                  <input
                    className="form-control"
                    defaultValue={15}
                    name={undefined}
                    onChange={[Function]}
                    placeholder="Frequency (minutes)"
                    type="number"
                    value={15} />
                </pure(Component)>
              </div>
            </Group>
          </pure(Component)>
          <pure(Component)
            label="Start time"
            onChange={[Function]}
            type="time"
            value="08:00">
            <Group
              label="Start time"
              onChange={[Function]}
              type="time"
              value="08:00">
              <div
                className="form-group">
                <label>
                  Start time
                </label>
                <pure(Component)
                  label="Start time"
                  onChange={[Function]}
                  type="time"
                  value="08:00">
                  <input
                    className="form-control"
                    defaultValue="08:00"
                    name={undefined}
                    onChange={[Function]}
                    placeholder="Start time"
                    type="time"
                    value="08:00" />
                </pure(Component)>
              </div>
            </Group>
          </pure(Component)>
          <pure(Component)
            label="End time"
            onChange={[Function]}
            type="time"
            value="16:00">
            <Group
              label="End time"
              onChange={[Function]}
              type="time"
              value="16:00">
              <div
                className="form-group">
                <label>
                  End time
                </label>
                <pure(Component)
                  label="End time"
                  onChange={[Function]}
                  type="time"
                  value="16:00">
                  <input
                    className="form-control"
                    defaultValue="16:00"
                    name={undefined}
                    onChange={[Function]}
                    placeholder="End time"
                    type="time"
                    value="16:00" />
                </pure(Component)>
              </div>
            </Group>
          </pure(Component)>
        </div>
      </TimetableEntry>
      <pure(Component)
        label="Speed (km/h)"
        onChange={[Function]}
        value={40}>
        <Group
          label="Speed (km/h)"
          onChange={[Function]}
          value={40}>
          <div
            className="form-group">
            <label>
              Speed (km/h)
            </label>
            <pure(Component)
              label="Speed (km/h)"
              onChange={[Function]}
              type="number"
              value={40}>
              <input
                className="form-control"
                defaultValue={40}
                name={undefined}
                onChange={[Function]}
                placeholder="Speed (km/h)"
                type="number"
                value={40} />
            </pure(Component)>
          </div>
        </Group>
      </pure(Component)>
      <pure(Component)
        label="Dwell time (seconds)"
        onChange={[Function]}
        value={10}>
        <Group
          label="Dwell time (seconds)"
          onChange={[Function]}
          value={10}>
          <div
            className="form-group">
            <label>
              Dwell time (seconds)
            </label>
            <pure(Component)
              label="Dwell time (seconds)"
              onChange={[Function]}
              type="number"
              value={10}>
              <input
                className="form-control"
                defaultValue={10}
                name={undefined}
                onChange={[Function]}
                placeholder="Dwell time (seconds)"
                type="number"
                value={10} />
            </pure(Component)>
          </div>
        </Group>
      </pure(Component)>
    </div>
  </section>
</Timetable>
`;
